src/coreclr/jit/morphblock.cpp

```diff
@@ -1595,6 +1593,12 @@ GenTree* Compiler::fgMorphInitBlock(GenTree* tree)
 //
 GenTree* Compiler::fgMorphStoreDynBlock(GenTreeStoreDynBlk* tree)
 {
+    if (!tree->Data()->OperIs(GT_CNS_INT, GT_INIT_VAL))
+    {
+        // Data is a location.
+        tree->Data()->gtFlags |= GTF_DONT_CSE;
```

Comment:
If we're relying on this NO_CSE, seems prudent to set it in import as well.

Reply:
I decided to leave this here only, I think we should either set it on all locations in the importer or none of them, and today we don't set it on any of them. As you probably know we rely on this flag for the similar purpose in `fgMorphLocalVar` as well.

I guess today only the IND form of locations is allowed for `STORE_DYN_BLK`, but you could imagine if we allowed locations in general then something like `STORE_DYN_BLK(LCL_VAR int, LCL_VAR int, 4)` would have also been incorrectly handled without this.